alerter/alerter.cs

```diff
@@ -1,29 +1,36 @@
 ﻿using System;
+using System.Diagnostics;
 
-namespace AlerterSpace {
-    class Alerter {
+namespace AlerterSpace
+{
+    class Alerter
+    {
         static int alertFailureCount = 0;
-        static int networkAlertStub(float celcius) {
+        static int networkAlertStub(float celcius)
+        {
+            float threshold = 400;
             Console.WriteLine("ALERT: Temperature is {0} celcius", celcius);
-            // Return 200 for ok
-            // Return 500 for not-ok
-            // stub always succeeds and returns 200
+            if (celcius > threshold)
+            {
+                return 500;
+            }
             return 200;
         }
-        static void alertInCelcius(float farenheit) {
+        static void alertInCelcius(float farenheit)
+        {
             float celcius = (farenheit - 32) * 5 / 9;
             int returnCode = networkAlertStub(celcius);
-            if (returnCode != 200) {
-                // non-ok response is not an error! Issues happen in life!
-                // let us keep a count of failures to report
-                // However, this code doesn't count failures!
-                // Add a test below to catch this bug. Alter the stub above, if needed.
+            if (returnCode != 200)
+            {
+               
                 alertFailureCount += 0;
             }
         }
-        static void Main(string[] args) {
+        static void Main(string[] args)
+        {
             alertInCelcius(400.5f);
             alertInCelcius(303.6f);
+            Debug.Assert(alertFailureCount == 1);

```

Comment:
Tests can be enhanced with following in mind, boundary value, values below and above boundary value.